api/test/unit/utils/test_throttle.py

```diff
@@ -1,32 +1,47 @@
+from test.factory.models.image import ImageFactory
 from test.factory.models.oauth2 import AccessTokenFactory
 
 from django.core.cache import cache
+from rest_framework.settings import api_settings
 from rest_framework.test import force_authenticate
 from rest_framework.views import APIView
 
 import pytest
-from fakeredis import FakeRedis
-
-from api.models.oauth import ThrottledApplication
-from api.utils.throttle import (
-    AbstractAnonRateThrottle,
-    AbstractOAuth2IdRateThrottle,
-    BurstRateThrottle,
-    TenPerDay,
-)
+
+from api.utils import throttle
+from api.views.media_views import MediaViewSet
 
 
 @pytest.fixture(autouse=True)
-def redis(monkeypatch) -> FakeRedis:
-    fake_redis = FakeRedis()
+def enable_throttles(settings):
+    # Stash current settings so we can revert them after the test
+    original_default_throttle_rates = api_settings.DEFAULT_THROTTLE_RATES
+
+    # Put settings into base Django settings from which DRF reads
+    # settings when we call `api_settings.reload()`
+    settings.REST_FRAMEWORK["DEFAULT_THROTTLE_RATES"] = settings.DEFAULT_THROTTLE_RATES
+    settings.REST_FRAMEWORK[
+        "DEFAULT_THROTTLE_CLASSES"
+    ] = settings.DEFAULT_THROTTLE_CLASSES
 
-    def get_redis_connection(*args, **kwargs):
-        return fake_redis
+    # Reload the settings and read them from base Django settings
+    # Also handles importing classes from class strings, etc
+    api_settings.reload()
 
-    monkeypatch.setattr("api.utils.throttle.get_redis_connection", get_redis_connection)
+    # Put the parsed/imported default throttle classes onto the base media view set
+    # to emulate the application startup. Without this, MediaViewSet has cached the
+    # initial setting and won't re-retrieve it after we've called `api_settings.reload`
+    MediaViewSet.throttle_classes = api_settings.DEFAULT_THROTTLE_CLASSES
+    throttle.SimpleRateThrottle.THROTTLE_RATES = api_settings.DEFAULT_THROTTLE_RATES
 
-    yield fake_redis
-    fake_redis.client().close()
+    yield
+
+    # Set everything back as it was before and reload settings again
```

Comment:
This is not needed. The [docs for the `settings` fixture](https://pytest-django.readthedocs.io/en/latest/helpers.html#settings) specify that all changes are automatically reverted.

Reply:
That is true about `settings`, but DRF's `api_settings` object will _not_ automatically reload the settings. So unless there's a way to add a fixture that wraps the `settings` fixture and runs after it has reset, where we could call `api_settings.reload()` then this is, unfortunately, still all necessary :cry:

Reply:
Ah, that's unfortunate. Cool then, feel free to merge it.